fipy/viewers/mayaviViewer/mayaviDaemon.py

```diff
@@ -126,35 +127,51 @@ def run(self):
 
         self.cellsource = self.setup_source(self.cellfname)
         if self.cellsource is not None:
-            tmp = [out.cell_data.scalars for out in self.cellsource.outputs \
+            # Newer versions of mayavi (> 4.7?) store AssignAttribute objects
+            # in outputs, so this clumsy bit is to extract the underlying
+            # DataSet objects.
+            # This is a clear sign that we're using this completely wrong,
+            # but, eh, who cares?
+            cellsourceoutputs = [out if isinstance(out, tvtk.DataSet)
+                                 else out.trait_get()['output']
+                                 for out in self.cellsource.outputs]
+            tmp = [out.cell_data.scalars for out in cellsourceoutputs \
```

Comment:
I think that the section where the variables are all `facesomething` is exactly the same as the section with `cellsomething` so that could be one function / method that takes different arguments. Up to you.

Reply:
Because of Traits, doing that is really clunky, but ultimately better, I think